allennlp/data/dataset_readers/language_modeling.py

```diff
@@ -2,7 +2,7 @@
 
 from overrides import overrides
 
-from . import DatasetReader
+from .dataset_reader import DatasetReader
```

Comment:
I can do an "absolute imports" PR next, that should be quick.

I don't think that would have fixed this. The issue (I think) is that `dataset_readers/__init__.py` is itself importing from this file, so that trying to import from the `dataset_readers` module (which means importing its `__init__.py`) creates a circular reference that mypy can't deal with